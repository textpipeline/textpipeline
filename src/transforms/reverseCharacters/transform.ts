import { Transform } from '../types';

const transform: Transform = {
  name: 'Reverse',
  slug: 'reverse',
  inputTypeName: 'Text',
  outputTypeName: 'Reversed Text',
  project: undefined,
  defaultOutput: '',
  execute: async input =>
    input
      .split('')
      .reverse()
      .join(''),
};

export default transform;
