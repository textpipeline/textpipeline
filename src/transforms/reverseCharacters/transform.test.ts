import reverseCharacters from './transform';

describe('reverseCharacters', () => {
  it('has the correct attributes', () => {
    expect(reverseCharacters).toMatchSnapshot();
  });

  describe('execute', () => {
    it('converts empty to empty', () => expect(reverseCharacters.execute('')).resolves.toEqual(''));

    it('returns itself if the input is 1 character', () =>
      expect(reverseCharacters.execute('a')).resolves.toEqual('a'));

    it('reverses characters in a string', () => expect(reverseCharacters.execute('abc')).resolves.toEqual('cba'));

    it('reverses characters in a string that has spaces', () =>
      expect(reverseCharacters.execute('abc def')).resolves.toEqual('fed cba'));

    it('reverses characters in a string that has punctuation', () =>
      expect(reverseCharacters.execute('abc, def.')).resolves.toEqual('.fed ,cba'));
  });
});
