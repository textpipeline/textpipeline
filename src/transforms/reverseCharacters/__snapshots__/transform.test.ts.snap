// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reverseCharacters has the correct attributes 1`] = `
Object {
  "defaultOutput": "",
  "execute": [Function],
  "inputTypeName": "Text",
  "name": "Reverse",
  "outputTypeName": "Reversed Text",
  "project": undefined,
  "slug": "reverse",
}
`;
